gusto/eady_diagnostics.py

```diff
@@ -1,50 +1,31 @@
-from firedrake import FunctionSpace, Function, \
-    SpatialCoordinate
-from gusto.diagnostics import DiagnosticField, KineticEnergy, \
-    CompressibleKineticEnergy
+from firedrake import SpatialCoordinate
+from gusto.diagnostics import Energy, \
+    KineticEnergy, CompressibleKineticEnergy
 from gusto.forcing import exner
 
 
-class KineticEnergyV(DiagnosticField):
+class KineticEnergyV(Energy):
     name = "KineticEnergyV"
 
-    def field(self, mesh):
-        if hasattr(self, "_field"):
-            return self._field
-        self._field = Function(FunctionSpace(mesh, "DG", 0), name=self.name)
-        return self._field
-
     def compute(self, state):
         u = state.fields("u")
-        kineticv = 0.5*u[1]*u[1]
-        return self.field(state.mesh).interpolate(kineticv)
+        energy = self.kinetic(u[1])
+        return self.field(state.mesh).interpolate(energy)
 
 
-class CompressibleKineticEnergyV(DiagnosticField):
+class CompressibleKineticEnergyV(Energy):
     name = "CompressibleKineticEnergyV"
 
-    def field(self, mesh):
-        if hasattr(self, "_field"):
-            return self._field
-        self._field = Function(FunctionSpace(mesh, "DG", 0), name=self.name)
-        return self._field
-
     def compute(self, state):
         u = state.fields("u")
         rho = state.fields("rho")
-        kineticv = 0.5*rho*u[1]*u[1]
-        return self.field(state.mesh).interpolate(kineticv)
+        energy = self.kinetic(u[1], rho)
+        return self.field(state.mesh).interpolate(energy)
 
 
-class EadyPotentialEnergy(DiagnosticField):
+class EadyPotentialEnergy(Energy):
```

Comment:
Is this unique to the Eady problem? Isn't it just the incompressible Boussinesq potential energy?